xarray/core/datatree.py

```diff
@@ -93,14 +108,66 @@ def _coerce_to_dataset(data: Dataset | DataArray | None) -> Dataset:
     return ds
 
 
-def _check_for_name_collisions(
-    children: Iterable[str], variables: Iterable[Hashable]
+def _join_path(root: str, name: str) -> str:
+    return root.rstrip("/") + "/" + name
```

Comment:
I used pathlib for these kind of operations internally - creating [`NodePath`](https://github.com/pydata/xarray/blob/6c2d8c3389afe049ccbfd1393e9a81dd5c759f78/xarray/core/treenode.py#L26) objects and using the `/` operator to join segments.